src/Misc/Primes.jl

```diff
@@ -116,6 +116,8 @@ struct PrimesSet{T}
       end
       p = next_prime(p, false)
     end
+    mod = abs(mod)
+    val = abs(val)
```

Comment:
```suggestion
    val = mod(val, mod)
```
To make sure this works if the input values have mixed signs.